### PR TITLE
migration tests to switch between bidirectional/regular clusterpairs

### PR DESCRIPTION
**Pull request by Rohit-PX**
**What type of PR is this?**
>integration-test

**What this PR does / why we need it**:

**Does this PR change a user-facing CRD or CLI?**:
no

**Is a release note needed?**:
no

**Does this change need to be cherry-picked to a release branch?**:
no

### CODE REVIEW

File: test/integration_test/migration_test.go [hunk 1]
```diff
@@ -2032,3 +2044,39 @@ func serviceAndServiceAccountUpdate(t *testing.T) {
 	testResult = testResultPass
 	logrus.Infof("Test status at end of %s test: %s", t.Name(), testResult)
 }
+
+func scheduleClusterPairGeneric(t *testing.T, ctxs []*scheduler.Context,
+	appKey, instanceID, string,
+	defaultClusterPairDir, projectIDMappings string,
+	skipStoragePair, resetConfig, pairReverse bool) {
```
[the remark this answers is not in the document]
Comment (Rohit-PX) [reply test/integration_test/migration_test.go]: Fixed.

File: test/integration_test/migration_test.go [hunk 2]
```diff
@@ -1626,7 +1638,7 @@ func suspendMigrationTest(t *testing.T) {
 
 	logrus.Infof("Successfully verified suspend migration case")
 
-	validateAndDestroyMigration(t, ctxs, preMigrationCtx, true, false, true, false, false)
+	validateAndDestroyMigration(t, ctxs, "mysql-migration-schedule-interval-autosuspend", "mysql-1-pvc", preMigrationCtx, true, false, true, false, false, true)
```
[the remark this answers is not in the document]
Comment (Rohit-PX) [reply test/integration_test/migration_test.go]: Fixed.

File: test/integration_test/migration_test.go [hunk 3]
```diff
@@ -2032,3 +2044,39 @@ func serviceAndServiceAccountUpdate(t *testing.T) {
...
+	} else {
+		// create, apply and validate cluster pair specs
+		err = scheduleClusterPair(ctxs[0], skipStoragePair, true, defaultClusterPairDir, projectIDMappings, pairReverse)
+		require.NoError(t, err, "Error scheduling cluster pair")
+	}
+}
+
+func blowNamespacesForTest(t *testing.T, instanceID, appKey string) {
+	namespace := fmt.Sprintf("%s-%s", appKey, instanceID)
+	err := core.Instance().DeleteNamespace(namespace)
```
[the remark this answers is not in the document]
Comment (Rohit-PX) [reply test/integration_test/migration_test.go]: Fixed by calling explicitly the `setSourceKubeconfig()` function to avoid any edge cases.

File: test/integration_test/migration_test.go [hunk 4]
```diff
@@ -2032,3 +2044,39 @@ func serviceAndServiceAccountUpdate(t *testing.T) {
...
+	require.NoError(t, err, "failed to delete namespace %s on destintation cluster", namespace)
```
[the remark this answers is not in the document]
Comment (Rohit-PX) [reply test/integration_test/migration_test.go:2122]: Fixed.

**Comment by Rohit-PX** [comment]
```
        --- PASS: TestMigration/testMigration/deploymentTest (423.38s)
        --- PASS: TestMigration/testMigration/deploymentMigrationReverseTest (1012.56s)
        --- PASS: TestMigration/testMigration/statefulsetTest (373.33s)
        --- PASS: TestMigration/testMigration/statefulsetStartAppFalseTest (891.70s)
        --- PASS: TestMigration/testMigration/statefulsetRuleTest (403.32s)
        --- PASS: TestMigration/testMigration/preExecRuleMissingTest (556.93s)
        --- PASS: TestMigration/testMigration/postExecRuleMissingTest (547.98s)
        --- PASS: TestMigration/testMigration/disallowedNamespaceTest (558.07s)
        --- PASS: TestMigration/testMigration/failingPreExecRuleTest (557.02s)
        --- PASS: TestMigration/testMigration/failingPostExecRuleTest (315.88s)
        --- PASS: TestMigration/testMigration/labelSelectorTest (803.21s)
        --- PASS: TestMigration/testMigration/labelExcludeSelectorTest (804.06s)
        --- PASS: TestMigration/testMigration/intervalScheduleTest (636.68s)
        --- PASS: TestMigration/testMigration/dailyScheduleTest (540.57s)
        --- PASS: TestMigration/testMigration/weeklyScheduleTest (551.80s)
        --- PASS: TestMigration/testMigration/monthlyScheduleTest (540.47s)
        --- PASS: TestMigration/testMigration/scheduleInvalidTest (269.24s)
        --- PASS: TestMigration/testMigration/intervalScheduleCleanupTest (805.28s)
        --- PASS: TestMigration/testMigration/networkpolicyTest (1209.07s)
        --- PASS: TestMigration/testMigration/endpointTest (601.47s)
        --- PASS: TestMigration/testMigration/clusterPairFailuresTest (276.47s)
        --- PASS: TestMigration/testMigration/scaleTest (467.69s)
        --- PASS: TestMigration/testMigration/pvcResizeTest (914.37s)
        --- PASS: TestMigration/testMigration/transformResourceTest (660.57s)
        --- PASS: TestMigration/testMigration/suspendMigrationTest (758.84s)
        --- PASS: TestMigration/testMigration/operatorMigrationMongoTest (334.99s)
        --- PASS: TestMigration/testMigration/operatorMigrationRabbitmqTest (282.78s)
        --- PASS: TestMigration/testMigration/bidirectionalClusterPairTest (161.28s)
        --- PASS: TestMigration/testMigration/serviceAndServiceAccountUpdate (1031.69s)
```

File: test/integration_test/migration_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (pp511) [reply test/integration_test/migration_test.go:2122]: missed it. can update in a later pr.

**Review by pp511 — APPROVED** [review]
lgt
